src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5563,7 +5563,16 @@ int wm_vuldet_collect_agent_software(scan_agent *agent, sqlite3 *db, scan_ctx_t*
 
                 if (agent->dist == FEED_MAC) {
                     wchr_replace(name, ' ', '_');
-                    if (vendor) wchr_replace(vendor, ' ', '_');
+                    // Some packages may come with a '@' followed up by the product's version.
+                    // Since we already have said version, we can discard it.
+                    // We are using the reverse variant just in case the package name contains a '@'.
+                    char* tmp = NULL;
+                    if (tmp = strrchr(name, '@'), tmp != NULL) {
+                        *tmp = '\0';
+                    }
+                    if (vendor) {
+                        wchr_replace(vendor, ' ', '_');
+                    }
```

Comment:
I understand the use of strrchr to avoid splitting a name that may have @ in its name, but given the case we have a name like this random@pkg@1.0, we are removing the version in the data provider functionality, and then here we are removing "@pkg" which interfere with the purpose of the function.

Reply:
If the name contains a @ character (I dont know if it is possible...). For example "random@pkg@1.0" the name captured would be "random".
Maybe a helper string function could be created that uses regex to detect this kind of "{package_name}@{version_number}" form. 
Maybe something like this one:
^(.+)@((?:[0-9]+\.)?(?:[0-9]+\.)?(?:[0-9]+))
The first capture group is the package name and the second capture group is version (supports major, major.minor and major.minor.patch forms)

Reply:
This section of the code was included just in case another provider (not `homebrew`) could have a version inserted in the name. 
But as you say, if the package name contains a `@` this will fail.

I'll remove these changes and leave only the fix in **syscollector.**